src/components/converter.tsx

```diff
@@ -0,0 +1,149 @@
+"use client";
+
+import { useState } from "react";
+import useSWR from "swr";
+
+const URL = process.env.NEXT_PUBLIC_URL;
+
+const encodedCredentials = btoa(
+  `${process.env.NEXT_PUBLIC_USERNAME}:${process.env.NEXT_PUBLIC_PASSWORD}`
+);
+
+export default function Converter() {
+  const fetcher = async (url) => {
+    const request = await fetch(`${url}`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Basic ${encodedCredentials}`,
+      },
+    });
+
+    const data = await request.json();
+    return data;
+  };
+
+  const handleConversion = async () => {
+    if (amount && from && to) {
+      try {
+        const result = await fetcher(
+          `https://xecdapi.xe.com/v1/convert_from/?from=${from}&to=${to}&amount=${amount}`
+        );
+
+        if (result) {
+          setResult(result.to[0].mid);
+        } else {
+          console.log("Error");
+        }
+      } catch (err) {
+        console.log("Error");
```

Comment:
Better error handling rather than console logging?

Reply:
@ArinzeJeffrey-droid. I am thinking of adding toast notifications to it.